// react-autocompleter/dist/ListItem-compiled.js
'use strict';

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol ? "symbol" : typeof obj; };

Object.defineProperty(exports, '__esModule', {
	value: true
});

var _createClass = function () {
	function defineProperties(target, props) {
		for (var i = 0; i < props.length; i++) {
			var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ('value' in descriptor) descriptor.writable = true;Object.defineProperty(target, descriptor.key, descriptor);
		}
	}return function (Constructor, protoProps, staticProps) {
		if (protoProps) defineProperties(Constructor.prototype, protoProps);if (staticProps) defineProperties(Constructor, staticProps);return Constructor;
	};
}();

var _get = function get(_x, _x2, _x3) {
	var _again = true;_function: while (_again) {
		var object = _x,
		    property = _x2,
		    receiver = _x3;_again = false;if (object === null) object = Function.prototype;var desc = Object.getOwnPropertyDescriptor(object, property);if (desc === undefined) {
			var parent = Object.getPrototypeOf(object);if (parent === null) {
				return undefined;
			} else {
				_x = parent;_x2 = property;_x3 = receiver;_again = true;desc = parent = undefined;continue _function;
			}
		} else if ('value' in desc) {
			return desc.value;
		} else {
			var getter = desc.get;if (getter === undefined) {
				return undefined;
			}return getter.call(receiver);
		}
	}
};

function _interopRequireDefault(obj) {
	return obj && obj.__esModule ? obj : { 'default': obj };
}

function _classCallCheck(instance, Constructor) {
	if (!(instance instanceof Constructor)) {
		throw new TypeError('Cannot call a class as a function');
	}
}

function _inherits(subClass, superClass) {
	if (typeof superClass !== 'function' && superClass !== null) {
		throw new TypeError('Super expression must either be null or a function, not ' + (typeof superClass === 'undefined' ? 'undefined' : _typeof(superClass)));
	}subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } });if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
}

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var ListItem = function (_Component) {
	_inherits(ListItem, _Component);

	function ListItem() {
		var _this = this;

		_classCallCheck(this, ListItem);

		_get(Object.getPrototypeOf(ListItem.prototype), 'constructor', this).apply(this, arguments);

		this.handleSelect = function () {
			_this.props.onSelect(_this.props.content);
		};
	}

	_createClass(ListItem, [{
		key: 'render',
		value: function render() {
			var _props = this.props;
			var content = _props.content;
			var active = _props.active;
			var className = _props.className;
			var styles = _props.styles;

			var classes = this.props.className;

			if (className && active) {
				classes += ' active';
			}

			return _react2['default'].createElement('li', { ref: 'autocompleter-listItems', className: classes, onMouseDown: this.handleSelect, style: styles }, content);
		}
	}], [{
		key: 'propTypes',
		value: {
			onSelect: _react.PropTypes.func,
			content: _react.PropTypes.string.isRequired,
			className: _react.PropTypes.string,
			styles: _react.PropTypes.object,
			active: _react.PropTypes.bool.isRequired
		},
		enumerable: true
	}]);

	return ListItem;
}(_react.Component);

exports['default'] = ListItem;
module.exports = exports['default'];
